infra/stream/test/TestFormatter.cpp

```diff
@@ -0,0 +1,241 @@
+#include "infra/stream/Formatter.hpp"
+#include "gtest/gtest.h"
+
+class TestFormatObject
+{
+public:
+    int v{ 9 };
+    bool b{ true };
+};
+template<>
+void infra::Formatter<TestFormatObject &>::Format(infra::StreamWriter& writer, infra::FormatSpec& spec)
+{
+    Formatter<decltype(value.v)>(value.v).Format(writer, spec);
+}
+
+class FormatTest : public testing::Test
+{
+public:
+    FormatTest()
+        : writer(streamWriter.Writer())
+    {}
+protected:
+    void AssertFormatter(const char* expected) const
+    {
+        EXPECT_EQ(expected, streamWriter.Storage());
+    }
+    infra::StringOutputStream::WithStorage<60> streamWriter;
+    infra::StreamWriter& writer;
+};
+TEST_F(FormatTest, emptyString)
+{
+    Format(writer, "");
+    AssertFormatter("");
+}
+TEST_F(FormatTest, simpleString)
+{
+    Format(writer, "a");
+    AssertFormatter("a");
+}
+TEST_F(FormatTest, missingArgument)
+{
+    Format(writer, "{}");
+    AssertFormatter("}");
+}
+TEST_F(FormatTest, outOfIndex)
+{
+    Format(writer, "{1}");
+    AssertFormatter("}");
+}
+TEST_F(FormatTest, stringWidth)
+{
+    Format(writer, "{:5}", "b");
+    AssertFormatter("b    ");
+}
+TEST_F(FormatTest, boolean)
+{
+    Format(writer, "{}", true);
+    AssertFormatter("true");
+}
+TEST_F(FormatTest, booleanCenter)
+{
+    Format(writer, "{:^14}", false);
+    AssertFormatter("    false     ");
+}
+
+TEST_F(FormatTest, integer0)
+{
+    Format(writer, "{0}", 0);
+    AssertFormatter("0");
+}
+TEST_F(FormatTest, uint8_t)
+{
+    uint8_t v = 255;
+    Format(writer, "{}", v);
+    AssertFormatter("255");
+}
+TEST_F(FormatTest, int8_t)
+{
+    int8_t v = -128;
+    Format(writer, "{}", v);
+    AssertFormatter("-128");
+}
+TEST_F(FormatTest, uint16_t)
+{
+    uint16_t v = 0xFFFF;
+    Format(writer, "{}", v);
+    AssertFormatter("65535");
+}
+TEST_F(FormatTest, int16_t)
+{
+    int16_t v = -0x8000;
+    Format(writer, "{}", v);
+    AssertFormatter("-32768");
+}
+/*
+TEST_F(FormatTest, uint32_t)
+{
+    uint32_t v = 0xFFFFFFFF;
+    Format(stream, "{}", v);
+    assertFormatter("4294967295");
+}
+TEST_F(FormatTest, int32_t)
+{
+    int32_t v = std::numeric_limits<int32_t>::min();
+    Format(stream, "{}", v);
+    assertFormatter("-2147483648");
+}
+TEST_F(FormatTest, uint64_t)
+{
+    uint64_t v = 0xFFFFFFFFFFFFFFFF;
+    Format(stream, "{}", v);
+    assertFormatter("18446744073709551615");
+}
+TEST_F(FormatTest, int64_t)
+{
+    int64_t v = std::numeric_limits<int64_t>::min();
+    Format(stream, "{}", v);
+    assertFormatter("-9223372036854775808");
+}
+*/
+TEST_F(FormatTest, integer12)
+{
+    Format(writer, "{0}", 12);
+    AssertFormatter("12");
+}
+TEST_F(FormatTest, integer12WidthZeroPadding)
```

Comment:
I seem to be missing coverage for width and padding with negative numbers. Am I correct that this needs extra tests?